### PR TITLE
[PRO 293] resource details route and page

### PR DESCRIPTION
**Pull request by yuuks**
Changes
---
* Adds a `get(id: string)` function to api/resources.ts for fetching an individual resource.
* Adds a `ResourceView` page component and route to router.
* Renders a `ResourceCard` to display resource info in `ResourceView`.

Testing
---
* Check out the [live preview](https://deploy-preview-117--fabulous-sprite-a61e48.netlify.app/)
* Navigate to `resources/:id` (replace id with number). Verify the screen looks good and up voting/down voting works.
* The rest of the site/navigation is working properly.

### CODE REVIEW

File: src/pages/ResourceView.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    setResource(newResource)
+  }
+
+  return (
+    <div>
+      <a role="button" onClick={() => navigate(-1)}>
+        <i className="bi bi-chevron-left align-middle" />
+        {t('ui.back')}
+      </a>
+      <div className="vertical-rhythm">
```
Comment (tasandberg) [inline src/pages/ResourceView.tsx:26]: I think if you remove this div and add the `vertical-rhythm` className to the top-level div, it'll sort out the spacing issue.

File: src/api/resources.ts [hunk 2]
```diff
@@ -35,3 +35,12 @@ export async function dislike(id: number) {
 
   return result
 }
+
+export async function get(id: number) {
```
Comment (tasandberg) [inline src/api/resources.ts:39]: There's a type mismatch between this expecting a number and the loader casting the resourceId as string, which is causing the preview build to fail when it transpiles the Typescript.

File: src/pages/ResourceView.tsx (hunk 1 above)
Comment (yuuks) [reply src/pages/ResourceView.tsx]: I've done that but it still seems to not add any spacing between the back button and the resource card.

**Review by tasandberg — CHANGES_REQUESTED** [review]
Just the fix on the back link spacing and this will be good to go. I noticed a bug where the upvote/downvote icons weren't filled in on page load even if the current user had voted on it. It's an issue with the API that I'll open a PR for.

File: src/pages/ResourceView.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export default function ResourceView() {
+  const navigate = useNavigate()
+  const { t } = useTranslation()
+  const data = useLoaderData() as ResourceLoaderReturn
+  const [resource, setResource] = useState(data.resource)
+  const onUpdateResource = (updatedResourceData: { resource: Resource }) => {
+    const updatedResource = updatedResourceData.resource
+    const newResource = { ...resource, ...updatedResource }
+
+    setResource(newResource)
```
Comment (tasandberg) [inline src/pages/ResourceView.tsx:17]: Might be nice to simplify to one line and save the variable declarations.
```suggestion
  setResource({ ...resource, ...updatedResourceData.resource })
```

Comment (yuuks) [reply src/pages/ResourceView.tsx]: Updated in latest commit 👍

File: src/pages/ResourceView.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  return (
+    <div className="vertical-rhythm">
+      <a role="button" onClick={() => navigate(-1)}>
+        <i className="bi bi-chevron-left align-middle" />
+        {t('ui.back')}
+      </a>
```
[the remark this answers is not in the document]
Comment (yuuks) [reply src/pages/ResourceView.tsx]: Thanks, looks good after that change.

**Review by tasandberg — APPROVED** [review]
Great work!

**Pull request merged**